### ISSUE TITLE
Bug: "utils" package is incompatible with node 18

### ISSUE THREAD

**Issue by sahariko** [typescript-eslint/typescript-eslint#6320]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

utils

### Repro Code

```typescript
// .eslintrc.js

module.exports = {
  // Other ESLint configuration
  plugins: ['@typescript-eslint'],
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  plugins: ['@typescript-eslint'],
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Linter can run

### Actual Result

Linter fails with the following error:

```
Error [ERR_PACKAGE_PATH_NOT_EXPORTED]: Failed to load plugin 'jest' declared in 'package.json » REDACTED': Package subpath './lib/definition' is not defined by "exports" in REDACTED/node_modules/eslint-scope/package.json
    at new NodeError (node:internal/errors:393:5)
    at throwExportsNotFound (node:internal/modules/esm/resolve:358:9)
    at packageExportsResolve (node:internal/modules/esm/resolve:668:3)
    at resolveExports (node:internal/modules/cjs/loader:529:36)
    at Module._findPath (node:internal/modules/cjs/loader:569:31)
    at Module._resolveFilename (node:internal/modules/cjs/loader:981:27)
    at Module._load (node:internal/modules/cjs/loader:841:27)
    at Module.require (node:internal/modules/cjs/loader:1061:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (REDACTED/node_modules/@typescript-eslint/utils/dist/ts-eslint-scope/Definition.js:4:22) {
  code: 'ERR_PACKAGE_PATH_NOT_EXPORTED'
}
```

This package doesn't define the `lib/definition` path in [its `package.json` "exports" field](https://github.com/eslint/eslint-scope/blob/34ffedc9645f3e5bf2111f766931efb0ff33040f/package.json#L7-L13).

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.48.1` |
| `@typescript-eslint/utils`             | `5.48.1` |
| `TypeScript`                           | `4.9.4` |
| `ESLint`                               | `7.22.0` |
| `node`                                 | `18.12.1` |

**Comment by sahariko** [comment]
It seems like that for some reason I have `eslint-scope` installed with both version 5 and 7, and for some reason 7 is hoisted and this package doesn't include its own `node_modules` directory with the correct version.

**Comment by bradzacher** [comment]
Yes, this is not an issue with our tooling - we test against node 18 (and mostly develop against it) with no issue.

The dependency on `eslint-scope` is explicitly pinned to [a v5 version](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/utils/package.json#L47) so that we don't trigger this breakage.

In our next major we'll also be removing this entirely so we can properly support the new versions (https://github.com/typescript-eslint/typescript-eslint/issues/4953).

---

I'd suggest deleting your `node_modules` folder and doing a fresh install. This sort of issue is usually indicative of a bad install.
